aten/src/ATen/native/Convolution.cpp

```diff
@@ -119,6 +123,29 @@ auto ConvParams::view1d_as_2d() -> void {
   }
 }
 
+bool ConvParams::use_cpu_depthwise3x3_winograd(
+    const at::Tensor& input, const at::Tensor& weight) const
+{
+#ifdef __ARM_NEON__
+  // Currently only 3x3 depthwise convolutions on tensors of float are supported.
+  return (input.ndimension() == 4) &&
+         (input.size(1) == groups) &&
+         (weight.ndimension() == 4 ) &&
+         (weight.size(0) % input.size(1) == 0) &&
+         (weight.size(2) == 3) &&
+         (weight.size(3) == 3) &&
+         (input.device().type() == c10::DeviceType::CPU) &&
+         (input.scalar_type() == at::kFloat) &&
+         (weight.device().type() == c10::DeviceType::CPU) &&
+         (weight.scalar_type() == at::kFloat) &&
+         !is_strided() &&
```

Comment:
I don't see anywhere in the implementation where you check the striding of the tensors.  Presumably you want to check that the tensors are contiguous or make them so?

Reply:
Thanks for pointing that out.